account/identity_relationship.go

```diff
@@ -0,0 +1,189 @@
+package account
+
+import (
+	"context"
+	"time"
+
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/gormsupport"
+	"github.com/fabric8-services/fabric8-auth/application/repository"
+	"github.com/fabric8-services/fabric8-auth/log"
+
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+type IdentityRelationship struct {
+	gormsupport.Lifecycle
+
+	ParentIdentity Identity `gorm:"primary_key"`
+	ChildIdentity Identity `gorm:"primary_key"`
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m IdentityRelationship) TableName() string {
+	return "identity_relationship"
+}
+
+// GetLastModified returns the last modification time
+func (m IdentityRelationship) GetLastModified() time.Time {
+	return m.UpdatedAt
+}
+
+// GormIdentityRelationshipRepository is the implementation of the storage interface for IdentityRelationship.
+type GormIdentityRelationshipRepository struct {
+	db *gorm.DB
+}
+
+// NewIdentityRelationshipRepository creates a new storage type.
+func NewIdentityRelationshipeRepository(db *gorm.DB) IdentityRelationshipRepository {
+	return &GormIdentityRelationshipRepository{db: db}
+}
+
+// IdentityRelationshipRepository represents the storage interface.
+type IdentityRelationshipRepository interface {
+	//repository.Exister
+	Load(ctx context.Context, ParentIdentity Identity, ChildIdentity Identity) (*IdentityRelationship, error)
+	Create(ctx context.Context, u *IdentityRelationship) error
+	Save(ctx context.Context, u *IdentityRelationship) error
+	List(ctx context.Context) ([]IdentityRelationship, error)
+	Delete(ctx context.Context, parentIdentityID uuid.UUID, childIdentityID uuid.UUID) error
+	Query(funcs ...func(*gorm.DB) *gorm.DB) ([]IdentityRelationship, error)
```

Comment:
I see there are Query() method in all repositories in this PR. Do you really need them everywhere? Are they all used?

Reply:
I'm actually thinking we should have more sensible "query" methods that reflect our likely use cases.  For example, it makes sense to be able to list the ResourceTypeScope records for a particular ResourceType, so we could add a ResourceTypeScope.List(ResourceType) method for that.  Same goes with listing Role objects for a particular ResourceType.  What do you think?

Reply:
Sure. We can add whatever methods we need. My point was mostly about not adding anything that we are not sure if we need now.